### PR TITLE
feat: add support for icu message format

### LINKED ISSUE #16: plural expressions can brake syntax highlighting
```ts
@Component(/* */)
export class SomeComponent {
  public list: string[];
}
```

```html
<div>
  {list.length, plural, =1 {One Item} other {{{list.length}} Items}}
</div>
<span>some mor text</span>
```

will look like this:

image

everything after the expression looses highlighting

### PR DESCRIPTION
**Pull request by jigfox**
fixes #16

### CODE REVIEW

File: examples/test.component.html [hunk 1]
```diff
@@ -56,6 +56,12 @@ <h1>Hi {{ user.name }}</h1>
   </div>
 </div>
 
+<div class="icu">
+  <span i18n
+    >Updated {minutes, plural, =0 {just now} =1 {one minute ago} other {{{minutes}}}
+  }</span>
```
Comment (dlvandenberg) [inline examples/test.component.html]:
```suggestion
  <span i18n>
     Updated {minutes, plural, =0 {just now} =1 {one minute ago} other {{{minutes}}}}
  </span>
```

File: grammar.js [hunk 2]
```diff
@@ -278,6 +278,27 @@ module.exports = grammar(HTML, {
...
+    icu: ($) =>
+      seq(
+        '{',
+        choice($._any_expression, $.concatenation_expression),
+        ',',
+        $.icu_clause,
+        ',',
+        repeat1($.icu_case),
+        '}',
+      ),
```
Comment (dlvandenberg) [inline grammar.js]:
```suggestion
    
    // -------- ICU expressions ---------
    icu: ($) =>
      seq(
        '{',
        choice($._any_expression, $.concatenation_expression),
        ',',
        $.icu_clause,
        ',',
        repeat1($.icu_case),
        '}',
      ),
      
    icu_clause: () => choice('plural', 'select'),

    icu_case: ($) => seq(
      $.icu_category,
      '{',
      repeat1($._node),
      '}'
    ),

    icu_category: () => /([=<>]\d+|other|[a-z][a-z0-9]+)/i,
```

Comment (dlvandenberg) [reply grammar.js]: Can you explain the regex in a comment perhaps?

**Review by dlvandenberg — CHANGES_REQUESTED** [review]

**Comment by dlvandenberg** [comment]
Thanks @jigfox , looking good :)

File: grammar.js (hunk 2 above)
Comment (jigfox) [reply grammar.js]: instead of commenting I simplified the regex, I hope this is fine

Comment (dlvandenberg) [reply grammar.js]: This regex allows too much I think. Perhaps just `/([=<>]\d+|[a-z]+)/i` would be better? This should cover the cases `=0`, `<0`, `>0`, `word`.

Comment (jigfox) [reply grammar.js]: I took a look into https://github.com/angular/angular/blob/edc1740d7e7f0296c719c7176c24031f55ae1487/packages/compiler/src/ml_parser/icu_ast_expander.ts and as far as I see, `>`, `<` aren't even allowed. But for `select`-clause I didn't see any restrictions

Comment (dlvandenberg) [reply grammar.js]: I see. Then the regex is fine.

**Review by dlvandenberg — APPROVED** [review]

**Pull request merged**